### PR TITLE
Add errWriter for fileexporter tests and remove testutils.LimitedWriter #3225

### PR DESCRIPTION
**Pull request by sethAmazon**
This ticket is currently for Phase2-GA-Roadmap

**Description:** Add Error Writer to be used in file exporter tests where limited writer was being used. Then remove limited writer from 

**Link to tracking Issue:** #3225 

**Testing:** Ran all file export tests in Intellij go test configuration. Run all unit test using make command.

### CODE REVIEW

File: testutil/testutil.go [hunk 1]
```diff
@@ -124,24 +124,16 @@ func createExclusionsList(exclusionsText string, t *testing.T) []portpair {
 	return exclusions
 }
 
-// LimitedWriter is an io.Writer that will return an EOF error after MaxLen has
-// been reached.  If MaxLen is 0, Writes will always succeed.
-type LimitedWriter struct {
-	bytes.Buffer
-	MaxLen int
+// ErrorWriter is an io.Writer that will return an error all ways
+type ErrorWriter struct {
```
Comment (bogdandrutu) [inline testutil/testutil.go:128]: This should be internal in the file_exporter_test.go  file.

**Review by bogdandrutu — CHANGES_REQUESTED** [review]

File: testutil/testutil.go (hunk 1 above)
Comment (sethAmazon) [reply testutil/testutil.go]: Resolved

File: exporter/fileexporter/file_exporter_test.go [hunk 2]
```diff
@@ -132,3 +127,17 @@ func tempFileName(t *testing.T) string {
 	require.NoError(t, os.Remove(socket))
 	return socket
 }
+
+// ErrorWriter is an io.Writer that will return an error all ways
+type ErrorWriter struct {
```
Comment (bogdandrutu) [inline exporter/fileexporter/file_exporter_test.go:132]: s/ErrorWriter/errWriter

No need to be public.

File: exporter/fileexporter/file_exporter_test.go [hunk 3]
```diff
@@ -132,3 +127,17 @@ func tempFileName(t *testing.T) string {
...
+
+func (e ErrorWriter) Write(p []byte) (n int, err error) {
+	return 0, errors.New("all ways return error")
+}
+
+func (e *ErrorWriter) Close() error {
+	return nil
+}
+
+var _ io.WriteCloser = new(ErrorWriter)
```
Comment (bogdandrutu) [inline exporter/fileexporter/file_exporter_test.go:143]: Can be removed, already used as WriteCloser in the exporter so it is guarantee that implements the interface.

File: exporter/fileexporter/file_exporter_test.go (hunk 2 above)
Comment (sethAmazon) [reply exporter/fileexporter/file_exporter_test.go]: Resolved

**Comment by bogdandrutu** [comment]
@sethAmazon do we have a PR in contrib to fix that usage (copy the code there)?

**Comment by sethAmazon** [comment]
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3501 sorry I did not understand what you were saying about copying until now. I have completed the pr here.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**